client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -30,11 +34,16 @@ import { countries } from 'components/phone-input/data';
 import { toIcannFormat } from 'components/phone-input/phone-number';
 import FormPhoneMediaInput from 'components/forms/form-phone-media-input';
 import wp from 'lib/wp';
+import ExtraInfoFrForm from 'components/domains/registrant-extra-info/fr-form';
+import config from 'config';
 
+const debug = debugFactory( 'calypso:my-sites:upgrades:checkout:domain-details' );
 const wpcom = wp.undocumented(),
 	countriesList = countriesListForDomainRegistrations();
 
-class DomainDetailsForm extends Component {
+class DomainDetailsForm extends PureComponent {
```

Comment:
✅ 

👍  for adding more information with this sort of subtlety.

I got surprised a couple of times and overreacted by binding everything with a `this`, but I agree with you - more meaning is good.  I think I've got it now: do bind anything you 
 pass to other components (e.g. event handlers), don't bind react lifecycle methods. If something complains about a missing this, bind that too ;)